### PR TITLE
Ability to force loading console commands from some plugins

### PR DESCRIPTION
**Pull request by mattab**
When Piwik is not installed yet

### CODE REVIEW

File: core/Console.php [hunk 1]
```diff
@@ -209,6 +209,35 @@ private function getDefaultPiwikCommands()
...
+
+        return $commands;
+    }
+
+    private function getCommandsFromPluginsMarkedInConfig()
+    {
+        $general = Config::getInstance()->getFromCommonConfig('General');
+
+        if (empty($general)) {
+            return null;
```
Comment (tsteur) [inline core/Console.php]: I haven't tested it but looks good I reckon. If you returned an empty array here and below you wouldn't need the check above for `!empty($commandsFromPluginsMarkedInConfig)` and code would be bit easier to read / cleaner. Ideally we always return the same type eg always `array`.

File: core/Console.php [hunk 2]
```diff
@@ -209,6 +209,35 @@ private function getDefaultPiwikCommands()
...
+        if(!empty($commandsFromPluginsMarkedInConfig)) {
+            $commands = array_merge($commands, $commandsFromPluginsMarkedInConfig);
+        }
+
+        return $commands;
+    }
+
+    private function getCommandsFromPluginsMarkedInConfig()
+    {
+        $general = Config::getInstance()->getFromCommonConfig('General');
```
Comment (tsteur) [inline core/Console.php]: Any reason we are not using `$config->General['always_load_commands_from_plugin']` and force it to be in common config?

**Comment by tsteur** [comment]
I haven't tested it but presume it works. Left 2 notes but not mandatory

File: core/Console.php (hunk 1 above)
Comment (mattab) [inline core/Console.php]: nice, had missed this one

File: core/Console.php (hunk 2 above)
Comment (mattab) [inline core/Console.php]: there was no actual reason and in the end code ends up much simpler :+1:

**Pull request merged**